packages/router/src/adapters/contract/contract.ts

```diff
@@ -273,121 +458,24 @@ export const prepareRouterContract = async (
-
-      // listen for event on contract
-      const { event } = await prepareEvt
-        .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
-        .waitFor(300_000);
-
-      // increment router fees
-      incrementRelayerFeesPaid(
-        chainId,
-        routerRelayerFee,
-        routerRelayerFeeAsset,
-        TransactionReasons.PrepareReceiver,
-        requestContext,
-      );
-
-      return await txService.getTransactionReceipt(chainId, event.transactionHash);
-    } catch (err: any) {
-      logger.warn("Router contract prepare: Gelato send failed", requestContext, methodContext, {
-        err: jsonifyError(err),
-      });
-    }
-  }
-
-  // 2. If gelato is not supported, or gelato send failed, try using the router network.
-  if (useRelayer) {
-    logger.info("Router contract prepare: sending using router network", requestContext, methodContext, {
-      prepareParams,
-      routerRelayerFeeAsset,
-      routerRelayerFee,
-    });
-
-    try {
-      const payload = {
-        chainId,
-        to: routerContractAddress,
-        type: MetaTxTypes.RouterContractPrepare,
-        data: {
-          params: prepareParams,
-          signature,
-          relayerFee: routerRelayerFee,
-          relayerFeeAsset: routerRelayerFeeAsset,
-        } as MetaTxPayloads[typeof MetaTxTypes.RouterContractPrepare],
-      };
-      await messaging.publishMetaTxRequest(payload);
-
-      // listen for event on contract
-      const { event } = await prepareEvt
-        .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
-        .waitFor(300_000);
-
-      // increment router fees
-      incrementRelayerFeesPaid(
-        chainId,
-        routerRelayerFee,
-        routerRelayerFeeAsset,
-        TransactionReasons.PrepareReceiver,
-        requestContext,
-      );
-
-      return await txService.getTransactionReceipt(chainId, event.transactionHash);
-    } catch (err: any) {
-      // NOTE: It is possible that the actual error was in the subscriber, and the above event's timeout
-      // (see waitFor) is the error we actually caught in this block.
-      logger.warn("Router contract prepare: router network failed", requestContext, methodContext, {
-        err: jsonifyError(err),
-      });
-    }
-  }
-
-  // 3. If all of the above failed or was otherwise not supported, use txservice to send the transaction.
-  logger.info("Router contract prepare: sending using txservice", requestContext, methodContext, { prepareParams });
-
-  const receipt = await txService.sendTx(
-    onchainTx,
+  return await sendRouterContractTx(
+    {
+      chainId,
+      routerContractAddress,
+      encodedData,
+      params: prepareParams,
+      signature,
+      relayerFee: useRelayer
+        ? {
+            asset: routerRelayerFeeAsset,
+            amount: routerRelayerFee,
+          }
+        : undefined,
+      reason: TransactionReasons.PrepareReceiver,
```

Comment:
Could still be a `metaTx` relay, as seen [here](https://github.com/connext/nxtp/blob/cdaccfdc3018803e2df6bba26e6b77f969de6582/packages/router/src/lib/operations/metaTx.ts#L156)